### ISSUE TITLE
Install lacks awareness of 32-bit and 64-bit

### ISSUE THREAD

**Issue by bruce-richardson** [mesonbuild/meson#4008]
When doing a 32-bit build on a 64-bit system, which AFAIK needs to use a cross-file, the resulting "ninja install" places the 32-bit libraries into the lib64 directory - possibly overriding proper 64-bit versions there. Probably for cross-builds "ninja install" should be disabled unless certain environment locations are explicitly given, e.g. DESTDIR for packaging, or "prefix" etc.

**Comment by nirbheek** [comment]
32-bit builds on 64-bit do not strictly need a cross-file, but restricting `ninja install` to not install into the system prefix sounds like a good idea.

**Comment by hanetzer** [comment]
```
meson build.lib64 && ninja -C build && DESTDIR=/tmp/foo ninja -C build.lib64 install
meson --libdir=lib build.lib32 && ninja -C build.lib32 && DESTDIR=/tmp/foo ninja -C build.lib32 install
```
^ Should do what you need. Depending on the distro and version, the 32 bit libdir may have
a different naming. On gentoo with a 17.1 profile 64 bit libs go into /usr/lib64 and 32 bit libs
go into /usr/lib, with no lib32 directory.

**Comment by dcbaker** [comment]
This should be fixed by b5d847e38c90292af351ff4aa99cef0300886660, which allows you to put libdir (and other install paths) in your cross file. Please re-open if that's not sufficient.
